syntax = "proto3";

package envoy.extensions.matching.common_actions.stats.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.matching.common_actions.stats.v3";
option java_outer_classname = "ActionsProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/matching/common_actions/stats/v3;statsv3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Stats actions]

message StatAction {
  // Action that drops the stat.
  message DropStatAction {}

  // Action that drops the tag.
  message DropTagAction {
    // The name of the tag.
    string target_tag_name = 1;
  }

  // Action that inserts a new tag, or replaces the tag value if the tag exists.
  message InsertTagAction {
    // The name of the tag.
    string tag_name = 1;

    // The value of the tag.
    string tag_value = 2;
  }

  oneof rule {
    DropStatAction drop_stat = 1;
    DropTagAction drop_tag = 2;
    InsertTagAction insert_tag = 3;
  }
}
